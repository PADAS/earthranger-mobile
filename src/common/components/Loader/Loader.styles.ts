import { StyleSheet } from 'react-native';
import { COLORS_LIGHT } from '../../constants/colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: '100%',
    width: '100%',
    backgroundColor: COLORS_LIGHT.blackTransparent88,
    alignItems: 'center',
    justifyContent: 'center',
  },
  activityIndicatorContainer: {
    height: 80,
    width: 80,
    borderRadius: 15,
    backgroundColor: COLORS_LIGHT.G6_LightGreyButton,
    alignItems: 'center',
    justifyContent: 'center',
  },
  activityIndicator: {
    flex: 1,
  },
});

export default styles;
