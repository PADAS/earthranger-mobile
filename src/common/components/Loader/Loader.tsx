// External Dependencies
import React from 'react';
import {
  ActivityIndicator,
  View,
  Modal,
} from 'react-native';
import { COLORS_LIGHT } from '../../constants/colors';

// Styles
import styles from './Loader.styles';

interface LoaderProps {
  isVisible: boolean;
}

const Loader = ({
  isVisible,
}: LoaderProps) => (
  <Modal visible={isVisible} transparent animationType="none">
    <View style={styles.container}>
      <View style={styles.activityIndicatorContainer}>
        <ActivityIndicator style={styles.activityIndicator} size="large" color={COLORS_LIGHT.erTeal} animating />
      </View>
    </View>
  </Modal>
);

export { Loader };
