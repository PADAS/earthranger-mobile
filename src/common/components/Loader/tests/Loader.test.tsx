// External Dependencies
import React from 'react';
import { render } from '@testing-library/react-native';

// Internal Dependencies
import { Loader } from '../Loader';

describe('Loader', () => {
  it('matches snapshot', () => {
    const { toJSON } = render(
      <Loader isVisible />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
