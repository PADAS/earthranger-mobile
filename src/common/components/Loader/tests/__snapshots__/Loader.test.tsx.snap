// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000088",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3F3F3",
          "borderRadius": 15,
          "height": 80,
          "justifyContent": "center",
          "width": 80,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#008586"
        size="large"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</Modal>
`;
